src/Processors/QueryPlan/ReadFromRemote.cpp

```diff
@@ -429,6 +429,7 @@ void ReadFromParallelRemoteReplicasStep::initializePipeline(QueryPipelineBuilder
     {
         shuffled_pool = shard.pool->getShuffledPools(current_settings);
         shuffled_pool.resize(max_replicas_to_use);
+        coordinator->adjustParticipatingReplicasCount(max_replicas_to_use);
```

Comment:
It seems nothing prevents us from creating the coordinator right after `max_replicas_to_use` is defined. Let's create the coordinator after this `if ... else ...` block

Reply:
Thank you for reviewing the code. Perhaps creating a new coordinator after this if...else... block is not a good solution. 
Assuming that the unavailability of replicas is just a temporary network issue, changing the replicas_count could affect the locality of the cache. 
Additionally, the branch where max_replicas_to_use < shard.getAllNodeCount() is not frequently executed. Adjusting the participating_replicas_count in this branch has the minimal impact on performance. 
I'm not sure if my understanding is correct.

Reply:
See https://github.com/ClickHouse/ClickHouse/pull/66880/files#r1688308531

Reply:
Ok,thanks